packages/@aws-cdk/aws-ec2/lib/network-acl.ts

```diff
@@ -0,0 +1,411 @@
+import { Construct, IResource,  Resource, ResourceProps } from '@aws-cdk/core';
+import { CfnNetworkAcl, CfnNetworkAclEntry, CfnSubnetNetworkAclAssociation  } from './ec2.generated';
+import { IVpc } from './vpc';
+
+export interface INetworkACL extends IResource {
+  /**
+   * ID for the current Network ACL
+   * @attribute
+   */
+  readonly networkAclId: string;
+}
+
+/**
+ * A NetworkAclBase that is not created in this template
+ */
+abstract class NetworkAclBase extends Resource implements INetworkACL {
+public abstract readonly networkAclId: string;
+constructor(scope: Construct, id: string, props?: ResourceProps) {
+    super(scope, id, props);
+
+  }
+
+  public get uniqueId() {
+    return this.node.uniqueId;
+  }
+}
+
+export interface NetworkAclProps {
+  /**
+   * The name of the NetworkAcl.
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a networkAclName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly networkAclName?: string;
+  /**
+   * The VPC in which to create the NetworkACL.
+   */
+  readonly vpc: IVpc;
+}
+
+export class NetworkAcl extends NetworkAclBase {
```

Comment:
I have defined these as atomic constructs and to keep them as near to L2 as possible, I based a lot of this on the SecuityGroup existing construct,
A lot of these classes are very similar in intent , but have been implemented very differently in CFT